irrd/updates/validators.py

```diff
@@ -202,22 +231,28 @@ def _check_mntners(self, mntner_list: List[str], source: str) -> bool:
         self.keycert_obj_pk. Updates and checks self.passed_mntner_cache
         to prevent double checking of maintainers.
         """
-        for mntner_name in mntner_list:
-            if mntner_name in self._passed_cache or mntner_name in self._pre_approved:
-                return True
+        mntner_pk_set = set(mntner_pk_list)
+        mntner_objs: List[RPSLMntner] = [m for m in self._mntner_db_cache if m.pk() in mntner_pk_set and m.source() == source]
+        mntner_pks_to_resolve: Set[str] = mntner_pk_set - {m.pk() for m in mntner_objs}
 
-        query = RPSLDatabaseQuery().sources([source])
-        query = query.object_classes(['mntner']).rpsl_pks(mntner_list)
-        results = list(self.database_handler.execute_query(query))
-        mntner_objs: List[RPSLMntner] = [rpsl_object_from_text(r['object_text']) for r in results]  # type: ignore
+        if mntner_pks_to_resolve:
+            query = RPSLDatabaseQuery().sources([source])
+            query = query.object_classes(['mntner']).rpsl_pks(mntner_pks_to_resolve)
+            results = list(self.database_handler.execute_query(query))
+
+            retrieved_mntner_objs: List[RPSLMntner] = [rpsl_object_from_text(r['object_text']) for r in results]   # type: ignore
+            self._mntner_db_cache.update(retrieved_mntner_objs)
```

Comment:
The set may not correctly see two mntner objects as the same, as they are two different RPSLMntner objects, causing duplicates.